tests/analysis.spec.ts

```diff
@@ -386,4 +386,25 @@ describe('Functional test of analysis', () => {
       expect(bundle).toBeTruthy();
     });
   });
+
+  describe('analyzeScmProject', () => {
```

Comment:
Failure scenarios are covered in `report.spec.ts`, didn't consider adding any here valuable/worth it because this method does very little besides proxying `reportScm` and the happy path test should cover that logic.

Reply:
I was wrong, the report tests are not actually calling the backend currently, as we're mocking the entire `reportBundle` and `reportScm` methods.
It would be possible to have them actually trigger the test, and they would succeed and eventually fail when polling, but they would require the relevant FFs for both the pipeline org and for any org devs want to test locally/in dev with, so it's not really worth the trouble IMO. Also because of the retries on 500 codes the client is currently doing, they would add significant delays to the test runs.

I added tickets for implementing SCM Report E2E tests in the Orb and including the Orb in the pipeline for `code-agent` (https://snyksec.atlassian.net/browse/ZEN-737). That should cover our happy path E2E and ensure we don't mess up the CLI flows.
As for failure cases, I am mulling it over, but for now I don't have any good ideas that wouldn't complicate the test suites in this project for relatively little gain.

Reply:
Totally makes sense 👍